src/purchase-client.ts

```diff
@@ -0,0 +1,113 @@
+import { MuxySDK } from ".";
+import { CurrentEnvironment, forceType } from './util';
+import Config from './config';
+
+import { TwitchBitsProduct, TwitchBitsTransaction } from './twitch';
+
+export enum TransactionType {
+  Twitch,
+  Server,
+  Unknown
+}
+
+export interface PurchaseClient {
+  sdk: MuxySDK;
+
+  getProducts(): Promise<Product[]>;
+  purchase(identifier: string): void;
+  onUserPurchase(callback: <T>(data: T) => void): void;
```

Comment:
Rather than `callback` taking a generic argument, you should define a base `Transaction` interface (that conforms to the `TwitchBitsTransaction` type) and have that passed as the parameter to the callback.